Tests/test_PDBList.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2016 by Jacek Smietanski.  All rights reserved.
+# This file is part of the Biopython distribution and governed by your
+# choice of the "Biopython License Agreement" or the "BSD 3-Clause License".
+# Please see the LICENSE file that should have been included as part of this
+# package.
+
+"""Testing access to the PDB over the internet."""
+
+import contextlib
+import os
+import shutil
+import tempfile
+import unittest
+
+import requires_internet
+requires_internet.check()
+
+# We want to test this module:
+from Bio.PDB.PDBList import PDBList
+
+
+class TestPBDListGetList(unittest.TestCase):
+    """Test methods responsible for getting lists of entries."""
+
+    def test_get_recent_changes(self):
+        """Tests the Bio.PDB.PDBList.get_recent_changes method."""
+        pdblist = PDBList(obsolete_pdb="unimportant")  # obsolete_pdb declared to prevent from creating the "obsolete" directory
+        url = pdblist.pdb_server + '/pub/pdb/data/status/latest/added.pdb'
+        entries = pdblist.get_status_list(url)
+        self.assertIsNotNone(entries)
+
+    def test_get_all_entries(self):
+        """Tests the Bio.PDB.PDBList.get_all_entries method."""
+        pdblist = PDBList(obsolete_pdb="unimportant")  # obsolete_pdb declared to prevent from creating the "obsolete" directory
+        entries = pdblist.get_all_entries()
+        # As number of entries constantly grow, test checks if a certain number was exceeded
+        self.assertTrue(len(entries) > 100000)
+
+    def test_get_all_obsolete(self):
+        """Tests the Bio.PDB.PDBList.get_all_obsolete method."""
+        pdblist = PDBList(obsolete_pdb="unimportant")  # obsolete_pdb declared to prevent from creating the "obsolete" directory
+        entries = pdblist.get_all_obsolete()
+        # As number of obsolete entries constantly grow, test checks if a certain number was exceeded
+        self.assertTrue(len(entries) > 3000)
+
+
+class TestPDBListGetStructure(unittest.TestCase):
+    """Test methods responsible for getting structures."""
+
+    @contextlib.contextmanager
+    def make_temp_directory(self, dir):
```

Comment:
Even within function scope, I prefer not to overwrite built-ins. `directory` is an alternative.